app/code/community/Bolt/Boltpay/controllers/FeatureswitchesController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Bolt magento plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2019 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Class Bolt_Boltpay_FeatureswitchesController
+ */
+class Bolt_Boltpay_FeatureswitchesController
+    extends Mage_Core_Controller_Front_Action implements Bolt_Boltpay_Controller_Interface
+{
+    use Bolt_Boltpay_Controller_Traits_WebHookTrait;
+
+    /**
+     * Initializes Controller member variables
+     */
+    public function _construct()
+    {
+        //TODO: remove this method after adding signature support on bolt side
+        $this->requestMustBeSigned = false;
+    }
+
+    /**
+     * Received empty request from Bolt.
+     * Send to bolt API request to update feature switches
+     * Send response with status - success or failure
+     */
+    public function changedAction()
+    {
+        $updateResult = Mage::getSingleton("boltpay/featureSwitch")->updateFeatureSwitches();
+        if ($updateResult) {
+            $this->sendResponse(self::HTTP_OK, array('status' => 'success'));
+        } else {
+            $this->sendResponse(self::HTTP_UNPROCESSABLE_ENTITY, array('status' => 'failure'));
```

Comment:
Curious why 422? Is there another status we could use that's better?

Reply:
We use code 422 everywhere in the plugin and it looks like it also good for this situation. From RFC:
---
422 UNPROCESSABLE ENTITY
The server understands the content type of the request entity (hence a 415 Unsupported Media Type status code is inappropriate), and the syntax of the request entity is correct (thus a 400 Bad Request status code is inappropriate) but was unable to process the contained instructions.
---
The request contains the instruction "get feature switches from server" and we are unable to process this instruction for some reason.
But I am open to any suggestions.

Reply:
Got it. I was just a little curious.